test/Riok.Mapperly.Tests/Mapping/EnumTest.cs

```diff
@@ -332,4 +338,61 @@ public void EnumToStringMappingDisabledShouldUseToStringMapping()
         );
         TestHelper.GenerateMapper(source).Should().HaveSingleMethodBody("return (string)source.ToString();");
     }
+
+    [Theory]
+    [InlineData(EnumMappingStrategy.ByName, "Value1", "Value1, Value2")]
+    [InlineData(EnumMappingStrategy.ByName, "Value1, Value2", "Value1")]
+    [InlineData(EnumMappingStrategy.ByName, "Value1, Value2, Value3", "Value1")]
+    [InlineData(EnumMappingStrategy.ByName, "Value1", "Value1, Value2, Value3")]
+    [InlineData(EnumMappingStrategy.ByValue, "Value1", "Value1, Value2")]
+    [InlineData(EnumMappingStrategy.ByValue, "Value1, Value2", "Value1")]
+    [InlineData(EnumMappingStrategy.ByValue, "Value1, Value2, Value3", "Value1")]
+    [InlineData(EnumMappingStrategy.ByValue, "Value1", "Value1, Value2, Value3")]
+    public Task EnumToAnotherEnumByStrategyMissingValues(
+        EnumMappingStrategy enumMappingStrategy,
+        string sourceEnumValues,
+        string targetEnumValues
+    )
+    {
+        var source = TestSourceBuilder.Mapping(
+            "A",
+            "B",
+            TestSourceBuilderOptions.Default with
+            {
+                EnumMappingStrategy = enumMappingStrategy
+            },
+            "class A { public C Value { get; set; } }",
+            "class B { public D Value { get; set; } }",
+            $"enum C {{ {sourceEnumValues} }}",
+            $"enum D {{ {targetEnumValues} }}"
+        );
+
+        return TestHelper.VerifyGenerator(source, null, enumMappingStrategy, sourceEnumValues, targetEnumValues);

```

Comment:
Instead of using all values as parameters, I'd use an additional test parameter describing the test case, what it actually tests (eg. `by name with missing source value`).
Be reminded to remove unused snapshot files after updating the snapshot name.

Reply:
Fixed.